Made the original sample modular. It is now much easier to read, understa

SamuelMarks opened the pull request:
Made the original sample modular. It is now much easier to read, understand and extend.

SamuelMarks committed:
Made the original sample modular. It is now much easier to read, understand and extend.

galkahana commented:
Hi Sam, 
i took your code. the issue that you saw in linking was due to one of the functions having a declaration with one order of the parameters, and definition with another one. As a result the compiler OKed the compilation, however linking could not find the function. to correct this just change the order of the variables in the definition.

I corrected the code and merged it to the master branch. you can see it in https://github.com/galkahana/PDFWriterSamples/blob/master/Parsing/Parsing.cpp

in addition i formatted the code a bit to fit to the rest of the samples, added references where they are preferred over copying, and adapted to using UTF8 instead of wide chars, which is what the library now prefers as a result of adapting to Mac OSX. i think that if you take the code now it should work fine with your project

SamuelMarks commented:
Awesome, thank you very much for accepting my commit!

:]

Samuel Marks

Degree: Bachelor of Information Technology
Emphasis: Majoring in Software Technology

College: Macquarie University

On Tue, Oct 4, 2011 at 5:08 AM, gal kahana
reply@reply.github.com
wrote:

> Hi Sam,
> i took your code. the issue that you saw in linking was due to one of the functions having a declaration with one order of the parameters, and definition with another one. As a result the compiler OKed the compilation, however linking could not find the function. to correct this just change the order of the variables in the definition.
> 
> I corrected the code and merged it to the master branch. you can see it in https://github.com/galkahana/PDFWriterSamples/blob/master/Parsing/Parsing.cpp
> 
> in addition i formatted the code a bit to fit to the rest of the samples, added references where they are preferred over copying, and adapted to using UTF8 instead of wide chars, which is what the library now prefers as a result of adapting to Mac OSX. i think that if you take the code now it should work fine with your project
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/galkahana/PDFWriterSamples/pull/1#issuecomment-2275049
